lib/linguist/heuristics.yml

```diff
@@ -686,7 +686,7 @@ disambiguations:
 - extensions: ['.stl']
   rules:
   - language: STL
-    pattern: '\A\s*solid[\s\S]*^endsolid(?:$|\s)'
+    pattern: '\A\s*solid(?:$|\s)[\s\S]*^endsolid(?:$|\s)'
```

Comment:
`(?:$|\s)` amounts to `\s` in this context.

Reply:
I'm far from an regex expert so I'm open to suggestions. I want to match both of these cases as they're both valid (there's  no space at the end of the `solid` line in the second example): 

``` 
solid foo
  [...]
endsolid
```

```
solid
  [...]
endsolid
```

… but not `solidfoo` or `solid,`.